packages/react-server/README.md

```diff
@@ -5,3 +5,251 @@ This is an experimental package for creating custom React streaming server rende
+          },
+          pull: (controller): ?Promise<void> => {
+            startFlowing(request, controller);
+          },
+          cancel: (reason): ?Promise<void> => {
+            stopFlowing(request);
+            abort(request, reason);
+          },
+        },
+        // $FlowFixMe[prop-missing] size() methods are not allowed on byte streams.
+        {highWaterMark: 0},
+      );
+      resolve({prelude: stream});
+    }
+    const request = createPrerenderRequest(
+      model,
+      clientManifest,
+      onAllReady,
+      onFatalError,
+      options ? options.onError : undefined,
+      options ? options.identifierPrefix : undefined,
+      options ? options.onPostpone : undefined,
+      options ? options.temporaryReferences : undefined,
+      __DEV__ && options ? options.environmentName : undefined,
+      __DEV__ && options ? options.filterStackFrame : undefined,
+    );
+    startWork(request);
+  });
+}
+```
+
+## `Flight` Reference (Incomplete)
+
+### `createRequest(model, bundlerConfig, ...options): Request`
+
+The signature of this method changes as we evolve the project so this Readme will omit the specific signature but generally this function will produce a Request that represents the rendering of some React application (the model) along with implementation specific bundler configuration. Typically this configuration will tell the `Flight` implementation how to encode Client References in the serialized output
+
+The `RequestInstance` represents the render.
+
+Rendering does not actually begin until you call `startWork`
+
+### `createPrerenderRequest(model, bundlerConfig, ...options): Request`
+
+This is similar to `createRequest` but it alters some internal semantics for how errors and aborts are treated. It returns the same type as `createRequest`.
+
+### `startWork(request: Request): void`
+
+When passed a request this will initiate the actual render. It will continue until it completes
+
+### `startFlowing(request: Request, destination: Destination): void`
+
+a destination is whatever the implementation wants to use for storing the output of the render. In existing implementations it is either a Node stream or a Web stream. When you call `startFlowing` the request will write to the destination continuously whenever more chunks are unblocked, say after an async function has resolved and there is something new to serialize. You can implement streaming backpressure using `stopFlowing()`
+
+### `stopFlowing(request: Request): void`
+
+If you need to pause or permanently end the writing of any additional serialized output for this request you can call `stopFlowing(request)`. You may start flowing again after you've stopped. This is how you would implement backpressure support for streams for instance. It's important to note that stopping flowing is not going to stop rendering it simply causes the request to buffer any serialized chunks until they are requested again with `startFlowing()`.
```

Comment:
This is an implementation detail though. E.g. Fizz doesn't just simply buffer serialized chunks. It's likely Flight also would become more lazy in that regard.

Reply:
E.g. it would be the wrong take away that it would be better for a consumer to just manage its own buffer.

Reply:
https://github.com/facebook/react/pull/31466